### PR TITLE
Tests how native-image handles resources lookup

### PR DESCRIPTION
**Pull request by Karm**
Tests resources lookup behaviour, URL trailing slash trimming etc.

### CODE REVIEW

**Comment by Karm** [comment]
@Karm  Fix Windows CR LF

**Comment by Karm** [comment]
This is how it looks like when it fails:

```
[ERROR] Failures: 
[ERROR]   AppReproducersTest.resLocationsB:250->resLocations:340 Something changed in how the resources are handled. Check comments on https://github.com/oracle/graal/issues/4326 and https://github.com/quarkusio/quarkus/pull/22403. See discrepancies: 
Expected: 6:  NO_SLASH FOLDER, Actual: 6:  N/A
Expected: 7:  NO_SLASH FOLDER, Actual: 7:  N/A
Expected: 5:  NO_SLASH FOLDER, Actual: 5:  SLASH FOLDER
 ==> expected: <true> but was: <false>
[INFO] 
[ERROR] Tests run: 3, Failures: 1, Errors: 0, Skipped: 1
```

**Comment by Karm** [comment]
@zakkak Those failures are expected, JDK 17 +  record annotations + Mandrel 22 and JDK 17 + GDB

**Review by zakkak — APPROVED** [review]
LGTM, thanks @Karm!

Please have a look at my comments for a potential improvement and a copyright year fix :)

File: testsuite/src/it/java/org/graalvm/tests/integration/utils/BuildAndRunCmds.java [hunk 1]
```diff
@@ -195,6 +195,14 @@
                     "--name", ContainerNames.JFR_SMOKE_BUILDER_IMAGE.name + "-build",
                     BUILDER_IMAGE, "-H:+AllowVMInspection", "-jar", "target/timezones.jar", "target/timezones"}
             // @see JFRTest.java
+    }),
+    RESLOCATIONS(new String[][]{
+            new String[]{"mvn", "package"},
+            new String[]{"native-image", "--diagnostics-mode", "--initialize-at-build-time=", "--no-fallback",
```
[the remark this answers is not in the document]
Comment (Karm) [reply testsuite/src/it/java/org/graalvm/tests/integration/utils/BuildAndRunCmds.java]: Thx

**Comment by Karm** [comment]
Thx @zakkak, changes included, merging....

**Pull request merged**